pkg/webhook/admission/deployment/accelerator_injector.go

```diff
@@ -16,61 +16,19 @@ limitations under the License.
 package deployment
 
 import (
-	"context"
-	"encoding/json"
-	"net/http"
-
 	"github.com/kubeflow/kfserving/pkg/constants"
 	"github.com/kubeflow/kfserving/pkg/utils"
 
 	appsv1 "k8s.io/api/apps/v1"
-
-	"github.com/kubeflow/kfserving/pkg/webhook/third_party"
-	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
-	"sigs.k8s.io/controller-runtime/pkg/webhook/admission/types"
 )
 
-// Mutator is a webhook that injects incoming pods
-type Mutator struct {
-	Client  client.Client
-	Decoder types.Decoder
-}
-
 // These constants are used for detecting and applying GPU selectors
 const (
 	GkeAcceleratorNodeSelector = "cloud.google.com/gke-accelerator"
 	NvidiaGPUTaintValue        = "present"
 )
 
-// Handle decodes the incoming Pod and executes mutation logic.
-func (mutator *Mutator) Handle(ctx context.Context, req types.Request) types.Response {
```

Comment:
I created a single mutator that calls both injectors instead of creating two separate webhooks... @ellis-bigelow  is this the right pattern? how can we get good logs for this so its easy to debug issues with the webhook - to figure out what has run, what hasnt, and who did what...?

Reply:
I was thinking about this. I think it's acceptable otherwise we will be making multiple http calls in a row to do mutation.